flink-formats/flink-parquet/src/test/java/org/apache/flink/formats/parquet/ParquetColumnarRowInputFormatTest.java

```diff
@@ -294,6 +294,80 @@ void testProjectionReadUnknownField(int rowGroupSize) throws IOException {
                 });
     }
 
+    @ParameterizedTest
+    @MethodSource("parameters")
+    void testProjectionReadUnknownFieldAcrossFiles(int rowGroupSize) throws IOException {
+        int number = 1000;
+        List<RowData> records = new ArrayList<>(number);
+        for (int i = 0; i < number; i++) {
+            Integer v = i;
+            records.add(newRow(v));
+        }
+        // create parquet files for both legacy and new
+        // f7 and f99 don’t exist in legacy parquet file
+        // f99 doesn't exist in new parquet file
+        // create legacy parquet file
+        RowType legacyParquetType =
+                new RowType(
+                        IntStream.range(0, 6) // assume f7 not exist in legacy parquet file.
```

Comment:
```suggestion
                        IntStream.range(0, 6) // assume f7 and following columns do not exist in legacy parquet file.
```
Am I missing something here or should is the comment not reflecting the statement? :thinking: Don't we only create a `RowType` for the first 6 columns? :thinking: